src/colorizer/ColorizeCanvas.ts

```diff
@@ -178,12 +185,36 @@ export default class ColorizeCanvas {
     this.colorMapRangeMax = 0;
     this.currentFrame = 0;
 
+    this.overlay = new CanvasOverlay();
+    this.showScaleBar = false;
+    this.frameToCanvasScale = new Vector4(1, 1, 1, 1);
+
     this.render = this.render.bind(this);
     this.getCurrentFrame = this.getCurrentFrame.bind(this);
     this.setOutOfRangeDrawMode = this.setOutOfRangeDrawMode.bind(this);
+    this.updateScaling = this.updateScaling.bind(this);
   }
 
-  get domElement(): HTMLCanvasElement {
+  /**
+   * The DOM element containing the canvas and its overlay.
+   */
+  get domElement(): HTMLDivElement {
+    if (!this.canvasContainer) {
+      // Set up the canvas overlay as a sibling in the DOM layout
+      // by creating a dummy parent container.
+      this.canvasContainer = document.createElement("div");
+      this.canvasContainer.appendChild(this.renderer.domElement);
+      this.canvasContainer.appendChild(this.overlay.domElement);
+      this.canvasContainer.style.position = "relative";
+      this.overlay.domElement.style.position = "absolute";
+      this.overlay.domElement.style.left = "0";
+      this.overlay.domElement.style.top = "0";
+    }
+
+    return this.canvasContainer;
+  }
+
+  get canvasElement(): HTMLCanvasElement {
     return this.renderer.domElement;
   }
```

Comment:
An odd distinction to make, though I see why you have to. Though is the plan to eventually render `CanvasOverlay` offscreen and draw it on top of the original canvas? If so, would these two properties remain?

Reply:
Yeah, ideally `CanvasOverlay` can be rendered with `OffscreenCanvas` and the `domElement` getter could be removed.